fmt.sh

```diff
@@ -143,6 +143,9 @@ else
   run javafmt "${javafmt_args[@]:-}" ${java_files[@]:-}
 fi
 
+# check for typescript code style
+run prettier --write language-support/ts navigator/frontend
```

Comment:
This doesn't just check, but actually rewrites your files ([documentation](https://prettier.io/docs/en/cli.html)).

Formatting TS code should honor the same conditions as other language-specific formatters in this script (Scala, Haskell, Java, Starlark):
- if the script is invoked with `--test`, this should only check and not rewrite (CI uses this)
- if the script is invoked with `--diff`, either the check or the rewrite should only look at the changes since the current branch split away from `main` (to minimize the cost of formatting locally)

Reply:
Any reason not to extend the check to the entire repository?

Reply:
Performance of prettier is too bad for that, at least so far I've tested.

Reply:
I expect `prettier` will be fast enough if you limit it to files that are seen by `git`, e.g. by running `git ls-files | xargs prettier`.

Reply:
That's a good idea, thank you I will try that!

Reply:
This should be the behavior of `--diff`. It should run on the whole repo otherwise. CI runs with `--test` only.

Reply:
`git ls-files` lists all files in the repository except un-indexed or ignored files.